src/plugins/data/public/query/query_string/language_service/language_service.ts

```diff
@@ -42,7 +39,12 @@ export class LanguageService {
    * Registers default handlers for index patterns and indices.
    */
   private registerDefaultLanguages() {
-    const defaultEditor = createEditor(SingleLineInput, SingleLineInput, DQLBody);
+    const defaultEditor = createEditor(
+      SingleLineInput,
+      SingleLineInput,
+      [createDefaultLanguageReference()],
```

Comment:
i see from my earlier comment.

This is already a lot cleaner so def want this in but I think we can think bigger similar to @joshuali925 stuff with the query header extension being a container this could be the same. And I think the naming will help because i was a little confused before.

We could make this more generic. Will comment in my overall thread.